### ISSUE TITLE
Does not fill password on AWS login page

### ISSUE THREAD

**Issue by ribtoks** [keepassxreboot/keepassxc-browser#1970]
## Expected Behavior
After clicking on the "green KeePassXC icon" in the password field, password is filled in.

## Current Behavior
Login username _is_ entered, but when going to the 2nd screen with password, it does not work. When clicking on it, icon disappears, password is not entered.

If account is selected from browser's extension context menu, password is also not filled in.

## Possible Solution
Only when selecting "Fill password" from the context menu, it works.

## Steps to Reproduce (for bugs)
1. Go to https://aws.amazon.com/ and start logging in
2. Click on the KeePassXC icon in the Login field -> navigate to the 2nd step
3. Click on the KeePassXC icon in the Password field -> nothing happens

## Debug info
KeePassXC - 2.7.6
KeePassXC-Browser - 1.8.7
Operating system: Mac, Linux (both)
Browser: Firefox (latest and greatest)

**Comment by varjolintu** [comment]
I have no problems with the site. Make sure your entry URL is `https://signin.aws.amazon.com` or similar.

**Comment by ribtoks** [comment]
@varjolintu  Look, I have this on 2 different laptops - 1 with latest macOS and a personal one with Ubuntu and it **never** worked** for the last many years (maybe 4?) as I'm using this otherwise wonderful software bundle (KeePassXC + browser add-on). It's just this website, other websites I'm using work, including with multiple accounts per site.

![image](https://github.com/keepassxreboot/keepassxc-browser/assets/505555/1b836720-e8e4-4df6-bef2-69596173fae9)


![Peek 2023-08-26 08-36](https://github.com/keepassxreboot/keepassxc-browser/assets/505555/70683c6d-c799-493d-8db1-8a7a0bf0fdd7)

**Comment by varjolintu** [comment]
Do you have `Use predefined sites for compatibility.` enabled in your extension settings?

**Comment by ribtoks** [comment]
Yes, `Use predefined sites for compatibility.` option is enabled (I also can see `signin.aws.amazon.com` in the list of "predefined sites")

**Comment by varjolintu** [comment]
Can I get the debug logs? The same page works just fine for me. Have you tried to fill the password from the context menu?

**Comment by ribtoks** [comment]
As I mentioned in the issue description, "Fill password" from the context menu _does work_. Clicking on the icon _does not_. Looking at debug logs myself, is it possible that KeePassXC thinks this is already TOTP screen, not the password screen?

![image](https://github.com/keepassxreboot/keepassxc-browser/assets/505555/9ba9b399-fd74-405d-8c3a-57bf757c6a9c)

**Comment by ribtoks** [comment]
Don't want to attach another gif, but when I click "Sign in" after pre-selecting the username and I'm redirected to the page 2 where I need to enter the password, KeePassXC Browser add-on logs that empty array is found in the "Input fields found:"

**Comment by varjolintu** [comment]
How are these settings in your config?
<img width="816" alt="Screenshot 2023-08-26 at 9 53 11" src="https://github.com/keepassxreboot/keepassxc-browser/assets/24570482/6e839cc8-03be-4944-a485-00be21fb03b3">

That TOTP error is not relevant, although it's related to the settings above. I think this issue would definitely need some debugging, but for now I cannot reproduce this.

EDIT: I just noticed (and remembered) that the password page has a different URL from the login page. You should add `https://amazon.com` or `https://www.amazon.com` to the Additional URL's list of your entry.

**Comment by ribtoks** [comment]
Just double checked, password page has the same URL (https://signin.aws.amazon.com/signin) and it's not an `iframe`

**Comment by ribtoks** [comment]
Settings report
![image](https://github.com/keepassxreboot/keepassxc-browser/assets/505555/201e8e4d-2f23-4af2-857f-512d641f81a8)

**Comment by varjolintu** [comment]
Another thing you can try is:
- Your main entry has `https://www.amazon.com/` as URL.
- Clone the entry and use references for username and password.
- Change cloned entry's URL to `https://signin.aws.amazon.com`.

This should also work.

**Comment by ribtoks** [comment]
Sadly references trick did not work :(

They both were visible in the username field, but the password field behavior was the same - icon disappears.

Thank you for trying to help me, btw.

**Comment by varjolintu** [comment]
I'd also suggest you to try another browser if the same happens there.

**Comment by otonoton** [comment]
I have the exact same problem. Green icon disappears when you click it, but context menu fill in works.

**Comment by aplotor** [comment]
+1. i tried choosing a custom login field "Password" but it doesnt do anything either

**Comment by droidmonkey** [comment]
I've been logging into AWS without issue (root account) multiple times. Cannot replicate this at least on edge.

**Comment by varjolintu** [comment]
The page should work without any Custom Login Field selections.

**Comment by droidmonkey** [comment]
Cannot replicate any problems on AWS sign in, try using the new sign in UI (see banner on top)

**Comment by ribtoks** [comment]
I still every day has this problem with KeyPassXC and browser add-on and AWS. Recently I set up a new laptop from scratch and it's the same. I'm surprised nobody has any problems with it. Again, I love KeePassXC project and it helped me tremendously, but AWS login just plain simple does not work. All other websites that I use, do work.

**Comment by varjolintu** [comment]
And for some reason I haven't been able to reproduce this problem, ever. Of course I would fix it otherwise.
